### PR TITLE
Better naming for index and search classes

### PR DESCRIPTION
**Pull request by tteofili**
This PR aims at resolving #2066. 
The names of the indexing-related classes all start with `Index` whereas searching-related class names start with`Search`. 
The `IndexInvertedDenseVectors` and `SearchInvertedDenseVectors` have also been moved to `io.anserini.index` and `io.anserini.search` packages accordingly.
Runners' configurations within `pom.xml` have been changed to point to the right classes, however, for the sake of backward compatibility (also considering documentation), the names of the resulting commands have been kept the same; however we can change that if we prefer new naming scheme to be reflected.

### CODE REVIEW

File: src/main/java/io/anserini/ann/EvalInvertedDenseVectors.java [hunk 1]
```diff
@@ -53,7 +55,7 @@
 
 import static org.apache.lucene.search.BooleanClause.Occur.SHOULD;
 
-public class ApproximateNearestNeighborEval {
+public class EvalInvertedDenseVectors {
```
Comment (lintool) [inline src/main/java/io/anserini/ann/EvalInvertedDenseVectors.java:58]: I think the point was to get rid of the `io.anserini.ann` package... so maybe move this to `io.anserini.search` - and rename to `Evaluate`? Typically, we'd write eval util classes in Python...

File: src/main/java/io/anserini/util/IOUtils.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class IOUtils {
+
+  private IOUtils() {
+  }
+
+  public static Map<String, List<float[]>> readGloVe(File input) throws IOException {
```
Comment (lintool) [inline src/main/java/io/anserini/util/IOUtils.java:31]: I don't think this needs to be generalized and abstracted in its own class? Evaluation of Glove seems pretty outdated...

File: src/main/java/io/anserini/ann/EvalInvertedDenseVectors.java (hunk 1 above)
Comment (tteofili) [reply src/main/java/io/anserini/ann/EvalInvertedDenseVectors.java:58]: ok, if we're to remove the `ann` package we also need to move `FeatureVectorTokenizer`,  and both`fw` and `lexlsh' packages, e.g., into `io.anserini.analysis` (as they are tokenizers+filters in the end).

Comment (lintool) [reply src/main/java/io/anserini/ann/EvalInvertedDenseVectors.java:58]: Agreed.

**Review by lintool — APPROVED** [review]

**Pull request merged**